node/actors/sync_blocks/src/peers/tests.rs

```diff
@@ -825,49 +826,25 @@ async fn processing_invalid_sync_states() {
 
     let mut invalid_sync_state = test_validators.sync_state(1);
     invalid_sync_state.first_stored_block = test_validators.final_blocks[2].justification.clone();
-    let err = peer_states
-        .validate_sync_state(invalid_sync_state)
-        .unwrap_err();
-    let err = format!("{err:?}");
-    assert!(err.contains("first_stored_block"), "{err}");
+    assert!(peer_states.validate_sync_state(invalid_sync_state).is_err());
```

Comment:
Checking for substrings in tests is a bad practice, because it turns the tests into change detectors and is imprecise in the first place (especially in the case of dynamically constructed error messages, where there is no guarantee that the error message won't include the expected substring by accident). I somehow missed that during the review, sorry. Please use strongly typed errors if you want to distinguish them programmatically (it can be as simple as having an enum with each variant wrapping anyhow error).

Reply:
Right, will change to strongly typed error then.